mikupad.html

```diff
@@ -2157,13 +2344,228 @@
 	const [stoppingStrings, setStoppingStrings] = useSessionState('stoppingStrings', defaultPresets.stoppingStrings);
 	const [stoppingStringsError, setStoppingStringsError] = useState(undefined);
 	const [savedScrollTop, setSavedScrollTop] = useSessionState('scrollTop', defaultPresets.scrollTop);
-
+	const [modalState, setModalState] = useState({});
+	const [logitBias, setLogitBias] = useSessionState('logitBias', defaultPresets.logitBias);
+	const [logitBiasParam, setLogitBiasParam] = useState({});
+	const [logitBiasTemp, setLogitBiasTemp] = useState([]);
+	const [logitBiasSorted, setLogitBiasSorted] = useState([]);
+	const [logitBiasInput, setLogitBiasInput] = useState({power:"0",string:""});
 	const [contextLength, setContextLength] = useSessionState('contextLength', defaultPresets.contextLength);
 	const [memoryTokens, setMemoryTokens] = useSessionState('memoryTokens', defaultPresets.memoryTokens);
-
 	const [authorNoteTokens, setAuthorNoteTokens] = useSessionState('authorNoteTokens', defaultPresets.authorNoteTokens);
 	const [authorNoteDepth, setAuthorNoteDepth] = useSessionState('authorNoteDepth', defaultPresets.authorNoteDepth);
 
+	const handleLogitBiasInput = (key,value) => {
+		setLogitBiasInput((prevLogitBiasInput) => {
+			return {
+				...prevLogitBiasInput,
+				[key]: value
+			}
+		});
+
+	}
+
+	const logitBiasAdd = async (biasPower="",biasString="",origValue="") => {
+		setLastError(undefined)
+		// abort if no input or power is NaN
+		if(!biasString) {return}
+		if (isNaN(+biasPower) || biasPower == "") { 
+			setLastError("Error: Bias is NaN")
+			return
+		}
+		biasPower = Number(biasPower)
+
+		const modBias = logitBias.bias;
+		
+		// delete entry if power 0 or empty
+		if (biasPower == 0) {	
+			if (!logitBias.bias[biasString])
+				return

```

Comment:
I think it would be helpful to warn the user that bias 0 does nothing.
```suggestion
			if (!logitBias.bias[biasString]) {
				setLastError("Error: Bias is 0")
				return
			}
```